Countries dropdown, shareable URL link, fetch arXiv and doi, enhance article UI, reusable code, unit tests using jest, rewrite entire registration

dinakar17 opened the pull request:
1. In the createcommunity page, I’ve implemented two things
i) Again improved code reusability by replacing input fields with reusable components Input.jsx and TextareaField.jsx.
ii) Added countries select dropdown using react-tailwindcss-select library for selecting the location.
![a5fa4bb4253ea33b708d89802f042637e7c213ab](https://github.com/m2b3/SciCommons-frontend/assets/85105044/7c76d129-d365-4644-91eb-c6b4bee6929a)

2. Added a shareable URL link for every comment (review) of the article.
![554ec03c87ebdaeb52425450aafd7284b0c7c844](https://github.com/m2b3/SciCommons-frontend/assets/85105044/f2d35bc4-1738-4951-a296-b2b73d3792f7)

3. Implemented the logic to fetch arXiv and doi articles and then add them to the database if that article hasn’t been added before. I’ve modified the logic to scrape the web data using DOMParser for arXiv. The Add Article button adds the article to the database and redirects the user to the article page. To achieve this functionality I’ve modified post request in the backend to send response body as well instead of plain text “Article added successfully”.
Also as demo depicts the same article can’t be added twice.
![955b47b6ff3d95ef6d1e783fc3ee77676d9fc5d7](https://github.com/m2b3/SciCommons-frontend/assets/85105044/501a70a8-bdee-4c41-8003-e0ce56f3adce)

4. Enhanced Article UI section design by adding tooltips (for clear message) and enhanced style to buttons, text, and other icons
![cce8fc787453f05be72808b52ec0f5e719c10d8e](https://github.com/m2b3/SciCommons-frontend/assets/85105044/0259594c-e819-4a6a-a6f3-0ff47fc9cac3)

5. Added two unit tests for InputField and Footer using jest and @react-test-library. Also configured babel for parsing jsx files while testing.
6. Redesigned the entire Registration with better code readability, reusability, and robust error-handling code.
![37e9f265a535c6ee9cf2a7405ff554dfe20de48c](https://github.com/m2b3/SciCommons-frontend/assets/85105044/e0b8a42d-8b50-4ca6-94bf-a2afc331eb20)

JyothiSwaroopReddy07 commented:
The comment Id part is already merged by some other contributor. It is breaking that functionality of scrolling view of nested comment. Your comment viewer only works for first layer of nested comments. 

Kindly remove the comment fetcher logic. It is already present on website.

[comment fetcher live site](https://www.scicommons.org/article/a645b597bfb644a99d16ea0f25157c37?commentId=16)

dinakar17 commented:
1. Modified the shareable URL link including better error handling logic and route feature.
![shreableurllinkfinaldemo](https://github.com/m2b3/SciCommons-frontend/assets/85105044/f0c83f82-1493-409b-9e72-c3107a14ed8e)
2. Added Skeleton Loaders for articles and communities pages so to enhance end user's UX experience
![skeletonloaders](https://github.com/m2b3/SciCommons-frontend/assets/85105044/c451db95-4062-4d7f-9e9e-a0b6ec404882)
3. Added Dark Mode functionality to the entire website. Changes in terms of TailwindCSS classes can be rolled over the time.

JyothiSwaroopReddy07 commented:
@dinakar17 Great PR, really good work
